app/search/dataone.py

```diff
@@ -147,6 +147,8 @@ def convert_result(self, result):
             # But there is also a named place available, which is what is being used here
             spatial_coverage=result.pop('placeKey', None),
             author = result.pop('author',None),
+            boundingbox = {'south': result.pop('southBoundCoord',None), 'north':result.pop('northBoundCoord',None),'west': result.pop('westBoundCoord',None), 'east':result.pop('eastBoundCoord',None)},
+            data_source = {'name': result.pop('datasource', None), 'url':'','logo':''},
```

Comment:
Also note that `datasource` contains the repository identifier, not, the repository name that is meant for display. In DataONE, you can get a mapping of repository metadata (identifier, name, description, logo URI, location, etc) for all repositories from the node capabilities service, e.g., `curl https://cn.dataone.org/cn/v2/node/urn:node:NPDC`, which produces:

```xml
<?xml version="1.0" encoding="UTF-8"?><?xml-stylesheet type="text/xsl" href="/cn/xslt/dataone.types.v2.xsl" ?>
<ns3:node xmlns:ns2="http://ns.dataone.org/service/types/v1" xmlns:ns3="http://ns.dataone.org/service/types/v2.0" replicate="false" synchronize="true" type="mn" state="up">
    <identifier>urn:node:NPDC</identifier>
    <name>Netherlands Polar Data Center (NPDC)</name>
    <description>The Netherlands Polar Data Center (NPDC) is part of the Netherlands Polar Program (NPP). NPDC archives and provides access to the data of Polar Research by researchers funded by Dutch Research Council (NWO) or otherwise carried out by researchers from Dutch universities and research institutions. The repository provides: 1) An overview of current and completed projects from the Netherlands Polar Programme (NPP) and other Dutch projects in the Polar Regions; 2) Access to the data of research carried out by Dutch researchers in the Polar Regions; and, 3) Links to external sources of Polar research data. For more information about the NPDC and the services it may offer to the Dutch Polar research community see https://npdc.nl/npdc.</description>
    <baseURL>https://sonode.dataone.org/NPDC</baseURL>
    <services>
        <service name="MNRead" version="v2" available="true"/>
        <service name="MNCore" version="v2" available="true"/>
        <service name="MNAuthorization" version="v2" available="false"/>
        <service name="MNStorage" version="v2" available="false"/>
        <service name="MNReplication" version="v2" available="false"/>
        <service name="MNView" version="v2" available="false"/>
        <service name="MNQuery" version="v2" available="false"/>
    </services>
    <synchronization>
        <schedule hour="*" mday="*" min="0,10,20,30,40,50" mon="*" sec="5" wday="?" year="*"/>
        <lastHarvested>2021-12-05T13:13:37.000+00:00</lastHarvested>
        <lastCompleteHarvest>1900-01-01T00:00:00.000+00:00</lastCompleteHarvest>
    </synchronization>
    <subject>urn:node:NPDC</subject>
    <contactSubject>http://orcid.org/0000-0002-6513-4996</contactSubject>
    <property key="CN_node_name">Netherlands Polar Data Center (NPDC)</property>
    <property key="CN_operational_status">operational</property>
    <property key="CN_date_operational">2021-09-10T20:19:42.0Z</property>
    <property key="CN_info_url">https://npdc.nl/</property>
    <property key="CN_logo_url">https://raw.githubusercontent.com/DataONEorg/member-node-info/master/production/graphics/web/NPDC.png</property>
    <property key="CN_location_lonlat">4.786,53.003</property>
</ns3:node>
```

Reply:
Thanks, @mbjones - that's a very neat way of getting that information.

@yemoski we definitely don't want to call that service for every query, but rather, we should use it and cache that data (perhaps with https://flask-caching.readthedocs.io/en/latest/index.html )

Reply:
sounds good!

Reply:
Just to be explicit, if you omit the repository identifier from that REST call, it will return the details for all currently registered repositories (rather than just one). This might be useful to you if you want to cache the info periodically rather than looking it up for specific repositories. This stuff doesn't tend to change very often once a repository is in place -- so a cache could be fairly long-lived and not miss much.